### ISSUE TITLE
QUANTILE aggregation function outputting wrong values

### ISSUE THREAD

**Issue by guilhermefarinha** [RediSearch/RediSearch#2503]
As a simple test to try the aggregate functions to compute quantiles from distributions, I have the following dataset:
```
[880000.0,
685000.0,
590000.0,
1200000.0,
1170000.0,
1145000.0,
3950000.0,
620000.0,
758000.0,
4850000.0,
800000.0,
340000.0,
530000.0,
500000.0,
540000.0,
2500000.0,
330000.0,
525000.0,
2500000.0,
350000.0,
590000.0,
1250000.0,
799000.0,
1380000.0]
```
which was uploaded directly to the **latest version** of redisearch in a field named `value`. Afterwards we were trying to compute the quantile 0.95 and 0.05 for this dataset using the following queries:
```
FT.AGGREGATE "test_quantile" * GROUPBY 0 REDUCE QUANTILE 2 @value 0.95 AS top_quantile
FT.AGGREGATE "test_quantile" * GROUPBY 0 REDUCE QUANTILE 2 @value 0.05 AS bottom_quantile
```
These two queries give the values `4850000` and `500000`. However, when one tries to compute these two quantiles outside of Redisearch using any standard statistical library, it outputs `3732500` and `341500.0`, respectively.

Are we doing anything wrong in the way we are constructing the aggregation queries?

**Comment by FVenancio** [comment]
I'm having the same issue!

**Comment by mariojfduarte** [comment]
This bug is also driving me insane!

**Comment by ashtul** [comment]
@guilhermefarinha @FVenancio @mariojfduarte 
We have a small bug that moved the result forward by two samples. This was fixed in #2507.
Please note, RediSearch uses a probabilistic data structure in order to support a large number of samples, and therefore there might be a minor skew of results.

Please note: There are multiple ways to calculate quantiles. [R has 9 types of algorithms](https://www.rdocumentation.org/packages/stats/versions/3.6.2/topics/quantile) to calculate quantile.
In RediSearch we always return a member of the set and in the case of the sample you gave, the result would be `3950000` and `340000` respectively.
